### PR TITLE
feat: add `insert()` method to editor API

### PR DESCRIPTION
**Pull request by ReFFaT**
I added the isertAt method, which adds to the cursor if the parameter is not passed, and to the index if it is passed. If the index is out of range, it uses the standard isert and prepend methods.

## Summary by Sourcery

Add support for inserting markup at a specific position or at the current cursor across editor implementations.

New Features:
- Expose an insertAt(markup, index?) API on the public Markdown editor and underlying editors to insert markup at the cursor or at a given absolute position.

Documentation:
- Document the new insertAt method on the public editor and content handler interfaces.

Tests:
- Add unit tests covering insertAt behavior for negative indices (prepend), in-range indices, out-of-range indices (append), and default cursor-based insertion in both Wysiwyg and markup editors.

### CODE REVIEW

File: packages/editor/src/common/index.ts [hunk 1]
```diff
@@ -13,4 +13,9 @@ export interface ContentHandler {
     prepend(markup: MarkupString): void;
     append(markup: MarkupString): void;
     moveCursor(position: 'start' | 'end'): void;
+    /**
+     * Insert markup at the current cursor position.
+     * If `index` is provided, insert at that position instead.
+     */
+    insertAt(markup: MarkupString, index?: number): void;
```
Comment (d3m1d0v) [inline packages/editor/src/common/index.ts]: Let's rename `insertAt() -> insert()` to be consistent with other methods.

And remove second `index` argument, as it works differently in different editor mods.

**Review by d3m1d0v — COMMENTED** [review]
And fix issues from sourcery

File: packages/editor/src/core/ContentHandler.test.ts [hunk 2]
```diff
@@ -158,4 +164,34 @@ describe('WysiwygContentHandler', () => {
         });
         expect(fn).toThrow();
     });
+
+    it('should insert inline at current cursor position', () => {
```
Comment (d3m1d0v) [inline packages/editor/src/core/ContentHandler.test.ts:173]: plz add test cases with replacement behavior

File: packages/editor/src/markup/editor.test.ts [hunk 3]
```diff
@@ -96,4 +96,14 @@ describe('MarkupContentHandler', () => {
         });
         expect(fn).toThrow();
     });
+
+    it('should insert at current cursor position', () => {
```
Comment (d3m1d0v) [inline packages/editor/src/markup/editor.test.ts:100]: plz add test cases with replacement behavior

File: packages/editor/src/core/ContentHandler.test.ts (hunk 2 above)
Comment (d3m1d0v) [reply packages/editor/src/core/ContentHandler.test.ts:173]: also, add test cases with block insertions (e.g. quote)

**Review by d3m1d0v — APPROVED** [review]
